packages/react-server-gulp-module-tagger/index.js

```diff
@@ -6,13 +6,19 @@ var replace = require("gulp-replace")
 //   - "__LOGGER__"
 //   - "__LOGGER__({ /* options */ })"
 var isWindows = ('win32' === process.platform)
-,   REPLACE_TOKEN = /(?:__LOGGER__|__CHANNEL__|__CACHE__)(?:\(\s*(\{[\s\S]*?\})\s*\))?/g
+,   DEFAULT_REPLACE_TOKEN = /(?:__LOGGER__|__CHANNEL__|__CACHE__)(?:\(\s*(\{[\s\S]*?\})\s*\))?/g
 ,   THIS_MODULE   = isWindows
 	? /(?:[^\\]+\\node_modules\\)?react-server-gulp-module-tagger\\index\.js$/
 	: /(?:[^\/]+\/node_modules\/)?react-server-gulp-module-tagger\/index\.js$/
 
 module.exports = function(config) {
 	config || (config = {});
+	var REPLACE_TOKEN;
+	if (config.token) {
+		REPLACE_TOKEN =  new RegExp("(?:" + config.token + ")(?:\\(\\s*(\\{[\\s\\S]*?\\})\\s*\\))?", 'g');
```

Comment:
Yeah, that's a good consideration.  I think in this case we can have our cake and eat it, too, though.  So long as `DEFAULT_REPLACE_TOKEN` stays at the package level where it is, the penalty for running it through the `RegExp` constructor will only be paid once.  After that it's just a compiled RegExp.  What we want to avoid for performance reasons is `new RegExp(...)` at _run time_.